grafana-data: vitest failed import in `fieldOverrides.mjs` for `react-use/lib/usePrevious`

### What happened?

Running `vitest`, few of my tests would break after setting `"type": "module",` in package.json

```
Error: Cannot find module '*/node_modules/react-use/lib/usePrevious' imported from */node_modules/@grafana/data/dist/esm/field/fieldOverrides.mjs
Did you mean to import "react-use/lib/usePrevious.js"?
```

### What did you expect to happen?

I expect my test to pass within esm context without making exceptions for a library 

### Did this work before?

I'm not sure, I'm currently migrating from `jest` to `vitest` on a new project.

### How do we reproduce it?

1. Create a new project with vite + vitest 
2. Add "type": "module",` in package.json
3. Setup a test component to use any import path dependent on `esm/field/fieldOverrides.mjs`


### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana:
OS: 
Browser:


### Grafana platform?

None

### Datasource(s)?

_No response_

--- a/packages/grafana-data/src/field/fieldOverrides.ts
+++ b/packages/grafana-data/src/field/fieldOverrides.ts
@@ -1,6 +1,6 @@
 import { isNumber, set, unset, get, cloneDeep } from 'lodash';
 import { useMemo, useRef } from 'react';
-import usePrevious from 'react-use/lib/usePrevious';
+import { usePrevious } from 'react-use';
 
 import { ThresholdsMode, VariableFormatID } from '@grafana/schema';
 

--- a/packages/grafana-sql/src/components/query-editor-raw/QueryValidator.tsx
+++ b/packages/grafana-sql/src/components/query-editor-raw/QueryValidator.tsx
@@ -1,7 +1,6 @@
 import { css } from '@emotion/css';
 import { useState, useMemo, useEffect } from 'react';
-import { useAsyncFn } from 'react-use';
-import useDebounce from 'react-use/lib/useDebounce';
+import { useAsyncFn, useDebounce } from 'react-use';
 
 import { formattedValueToString, getValueFormat, TimeRange } from '@grafana/data';
 import { Icon, Spinner, useTheme2 } from '@grafana/ui';

--- a/packages/grafana-sql/src/components/visual-query-builder/SQLWhereRow.tsx
+++ b/packages/grafana-sql/src/components/visual-query-builder/SQLWhereRow.tsx
@@ -1,4 +1,4 @@
-import useAsync from 'react-use/lib/useAsync';
+import { useAsync } from 'react-use';
 
 import { SelectableValue, TypedVariableModel } from '@grafana/data';
 import { getTemplateSrv } from '@grafana/runtime';

--- a/packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx
+++ b/packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx
@@ -3,7 +3,7 @@ import { autoUpdate, flip, offset, shift, useFloating } from '@floating-ui/react
 import Prism, { Grammar, LanguageMap } from 'prismjs';
 import { memo, useEffect, useRef, useState } from 'react';
 import * as React from 'react';
-import usePrevious from 'react-use/lib/usePrevious';
+import { usePrevious } from 'react-use';
 import { Value } from 'slate';
 import Plain from 'slate-plain-serializer';
 import { Editor } from 'slate-react';

--- a/packages/grafana-ui/src/components/DataLinks/DataLinkSuggestions.tsx
+++ b/packages/grafana-ui/src/components/DataLinks/DataLinkSuggestions.tsx
@@ -2,7 +2,7 @@ import { css, cx } from '@emotion/css';
 import { groupBy, capitalize } from 'lodash';
 import { useRef, useMemo } from 'react';
 import * as React from 'react';
-import useClickAway from 'react-use/lib/useClickAway';
+import { useClickAway } from 'react-use';
 
 import { VariableSuggestion, GrafanaTheme2 } from '@grafana/data';
 

--- a/packages/grafana-ui/src/components/Input/Input.tsx
+++ b/packages/grafana-ui/src/components/Input/Input.tsx
@@ -1,6 +1,6 @@
 import { css, cx } from '@emotion/css';
 import { forwardRef, HTMLProps, ReactNode, useContext } from 'react';
-import useMeasure from 'react-use/lib/useMeasure';
+import { useMeasure } from 'react-use';
 
 import { GrafanaTheme2 } from '@grafana/data';
 

--- a/packages/grafana-ui/src/components/Segment/SegmentAsync.story.tsx
+++ b/packages/grafana-ui/src/components/Segment/SegmentAsync.story.tsx
@@ -2,7 +2,7 @@ import { action } from '@storybook/addon-actions';
 import { Meta, StoryFn } from '@storybook/react';
 import { useState } from 'react';
 import * as React from 'react';
-import { AsyncState } from 'react-use/lib/useAsync';
+import { type AsyncState } from 'react-use/lib/useAsync';
 
 import { SelectableValue } from '@grafana/data';
 

--- a/packages/grafana-ui/src/components/Segment/SegmentAsync.tsx
+++ b/packages/grafana-ui/src/components/Segment/SegmentAsync.tsx
@@ -3,7 +3,7 @@ import { isObject } from 'lodash';
 import { HTMLProps } from 'react';
 import * as React from 'react';
 import { useAsyncFn } from 'react-use';
-import { AsyncState } from 'react-use/lib/useAsync';
+import { type AsyncState } from 'react-use/lib/useAsync';
 
 import { SelectableValue } from '@grafana/data';
 

--- a/packages/grafana-ui/src/components/Segment/SegmentInput.tsx
+++ b/packages/grafana-ui/src/components/Segment/SegmentInput.tsx
@@ -1,7 +1,7 @@
 import { cx, css } from '@emotion/css';
 import { HTMLProps, useRef, useState } from 'react';
 import * as React from 'react';
-import useClickAway from 'react-use/lib/useClickAway';
+import { useClickAway } from 'react-use';
 
 import { useStyles2 } from '../../themes';
 import { measureText } from '../../utils/measureText';

